### DISCUSSION TITLE
Ported SMON

### DISCUSSION THREAD

**Discussion by wolfmanjm** [picocomputer/community#78] [Q&A]
SMON is a full blown monitor with memory dumps, assembler and disassembler and most important single step debugging.

https://github.com/wolfmanjm/smon6502

**Comment by rumbledethumps** [comment]
Single stepping a 6502 is pure magic. This is a great way to begin learning assembly. Thanks.

**Comment by wolfmanjm** [comment]
As an interesting side note, I used Rake here to build (which is based on ruby a scripting language). Of note is the part where I scan the listing file for the addresses of the reset, irq and nmi code and extract those addresses and feed them into the rp6502.py call. This means I do not have to maintain them separately whenever the code changes.
I suspect this is easily done in python as well. I use Rake as I am not a fan of CMake or VSCode, I prefer command line tools and an editor.
This would also be easily adaptable to different assemblers, this one uses VASM.

**Comment by James-HoneyBadger** [comment]
flippin' awesome! much like the good old days of machine coding on the KIM-1 or other such single board computers
